### PR TITLE
Lab1

### PR DESCRIPTION
**Pull request by oleksandr-holan**

### CODE REVIEW

File: lab1/src/icecream.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+function isYesNo(input: string): input is YesNo {
+    return Object.values(YesNo).includes(input as YesNo);
+}
+
+let sizeInput: string | null;
+let toppingsInput: string | null;
+let toppingsArray: string[];
+let marshmallowInput: string | null;
+
+while (true) {
```
Comment (karachevtsev) [inline lab1/src/icecream.ts]: DRY, you have three separate while loops for user inputs (line 61, line 77, line 99). Create validation function for user input.

File: lab1/src/icecream.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    });
+
+    if (addMarshmallow === YesNo.Yes) {
+        cost += MarshmallowCost;
+    }
+
+    return cost;
+}
+
+function isSize(input: string): input is Size {
```
Comment (karachevtsev) [inline lab1/src/icecream.ts]: functions isSize, isTopping, isYesNo do the same thing, instead cereate helper function.

File: lab1/src/icecream.ts (hunk 1 above)
Comment (oleksandr-holan) [reply lab1/src/icecream.ts]: Split the code into getInput() func with a while loop and validation functions

File: lab1/src/icecream.ts (hunk 2 above)
Comment (oleksandr-holan) [reply lab1/src/icecream.ts]: created isValidEnumValue to assert the values and validateEnumValue, validateEnumArray to validate input

**Pull request merged**